fin_control/src/fin_control.cpp

```diff
@@ -193,46 +193,40 @@ void FinControl::reportAngles()
 {
   const char *log;
   fin_control::ReportAngle message;
-
-  if (!reportAnglesEnabled) return;  // do not interfere with fins updating
+  bool servos_angles_info_complete = true;
 
   message.header.stamp = ros::Time::now();
-  message.angle_in_radians = 0;
-
-  boost::mutex::scoped_lock lock(m_mutex);
-
-  for (int x = 0; x < numOfIDs; x++)
-  {
-    reportAngle(ids[x]);
-  }
-}
-void FinControl::reportAngle(uint8_t id)
-{
-  const char *log;
-  fin_control::ReportAngle message;
+  for (int i = 0; i < numOfIDs; i++) message.angles_in_radians.push_back(0.);
 
   if (!reportAnglesEnabled) return;  // do not interfere with fins updating
 
-  message.header.stamp = ros::Time::now();
-  message.angle_in_radians = 0;
-
-  message.ID = id;
-  // get from dynamixel
-  if (myWorkBench.getRadian(id, &message.angle_in_radians, &log))
+  for (int id = 0; id < numOfIDs; id++)
```

Comment:
@patapufate nit: post-fix operators may or may not incur additional overhead, so it's good practice to use prefix ones unless necessary:
```suggestion
  for (int id = 0; id < numOfIDs; ++id)
```

Reply:
Done